web/ui/react-app/src/utils/index.ts

```diff
@@ -259,3 +260,11 @@ export const parsePrometheusFloat = (value: string) => {
     return Number(value);
   }
 };
+
+export const checkReady = (res: FetchStateReady): boolean => {
+  const { ready, isLoading, isUnexpected } = res;
+  if (!ready && !isLoading && !isUnexpected) {
```

Comment:
I'm a bit confused by this string of AND conditions: So e.g. if it's not ready, but it's still loading, then we already return `true`?

Intuitively I would have expected something closer to:

```suggestion
  return ready && !isLoading && !isUnexpected;
```

Reply:
I think that function could be simplified, but my logic here is that we can only determine that the server is on but not ready if:

1. The variable we get indicates it's not ready
2. We have stopped loading.
3. The response is not an error.

If the response is an error, that means something is wrong with the server or it's not on, so we want to let those error messages show and return `true`. We could probably change this to be `showReadyScreen`, where we return true in the case that we can determine that we should show the ready screen, and false for everything else.

Reply:
Yeah, that would be more along my expectation - treat anything as not ready unless we positively get a working `ready` message back from Prometheus. Otherwise, the startup indicator component should be responsible for saying "nope it's not ready yet" and either show the loading state or the readiness-loading error.